org-code-javabuilder/playground/src/main/java/org/code/playground/Item.java

```diff
@@ -93,16 +95,28 @@ protected String getId() {
   }
 
   protected void sendChangeMessage(String key, String value) {
-    HashMap<String, String> details = this.getIdDetails();
-    details.put(key, value);
-    this.playgroundMessageHandler.sendMessage(
-        new PlaygroundMessage(PlaygroundSignalKey.CHANGE_ITEM, details));
+    if (this.shouldSendMessages) {
+      HashMap<String, String> details = this.getIdDetails();
+      details.put(key, value);
+      this.playgroundMessageHandler.sendMessage(
+          new PlaygroundMessage(PlaygroundSignalKey.CHANGE_ITEM, details));
+    }
```

Comment:
nit: minor potential cleanup - we can create a hashmap of just `key, value` and call `this.sendChangeMessage(details)` here so we only have to check `this.shouldSendMessages` in one method

Reply:
done!